File: src/test/java/com/telenordigital/prime/firebase/SubscriberImplTest.java
package com.telenordigital.prime.firebase;

import org.junit.Test;

import static org.junit.Assert.*;

public class SubscriberImplTest {

    private final SubscriberImpl fbs = new SubscriberImpl();

    @Test
    public void asMap() throws Exception {
        assertEquals(2, fbs.asMap().size());
        assertTrue(fbs.asMap().containsKey("noOfBytesLeft"));
        assertTrue(fbs.asMap().containsKey("msisdn"));

        assertEquals(0L, fbs.asMap().get("noOfBytesLeft"));
        assertEquals(null, fbs.asMap().get("msisdn"));
    }

    @Test
    public void getAndSetFbKey() throws Exception {
        assertEquals(null, fbs.getFbKey());
        final String fbkey = "foobar";
        fbs.setFbKey(fbkey);
        assertEquals(fbkey, fbs.getFbKey());
    }

    @Test
    public void getAndSetNoOfBytesLeft() throws Exception {
        assertEquals(0L, fbs.getNoOfBytesLeft());
        final long noOfBytesLeft = 123823838L;
        fbs.setNoOfBytesLeft(noOfBytesLeft);
        assertEquals(noOfBytesLeft, fbs.getNoOfBytesLeft());
    }

    @Test
    public void getAndSetMsisdn() throws Exception {
        assertEquals(null, fbs.getMsisdn());
        final String msisdn = "+47123456";
        fbs.setMsisdn(msisdn);
        assertEquals(msisdn, fbs.getMsisdn());
    }
}

File: prime/src/test/kotlin/org/ostelco/prime/firebase/SubscriberImplTest.kt
package org.ostelco.prime.firebase

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import org.ostelco.prime.firebase.entities.asMap
import org.ostelco.prime.model.Subscriber

class SubscriberImplTest {

    private val MSISDN = "+4712345678"
    private val NO_OF_BYTES_LEFT = 123823838L

    companion object {
        private const val NO_OF_BYTES_LEFT_KEY = "noOfBytesLeft"
        private const val MSISDN_KEY = "msisdn"
    }

    private val fbs = Subscriber(MSISDN, NO_OF_BYTES_LEFT)

    @Test
    fun asMap() {
        assertEquals(2, fbs.asMap().size.toLong())
        assertTrue(fbs.asMap().containsKey(NO_OF_BYTES_LEFT_KEY))
        assertTrue(fbs.asMap().containsKey(MSISDN_KEY))
        assertEquals(MSISDN, fbs.asMap()[MSISDN_KEY])
    }

    @Test
    fun getAndSetNoOfBytesLeft() {
        val noOfBytesLeft = 123823838L
        assertEquals(noOfBytesLeft, fbs.noOfBytesLeft)
    }

    @Test
    fun getAndSetMsisdn() {
        val msisdn = "+4712345678"
        assertEquals(msisdn, fbs.msisdn)
    }
}
